src/main/java/su/litvak/chromecast/api/v2/ChromeCast.java

```diff
@@ -86,6 +86,21 @@ public String getApplication() {
     public void setApplication(String application) {
         this.application = application;
     }
+    
+    /** Returns the {@link #channel} and opens it if it's not yet or
+     * no longer open.
+     * @return an open channel.
+     */
+    private Channel channel() throws IOException {
```

Comment:
Sorry for the delay. I think this can be done in a following way:

1) Introduce new parameter called `autoReconnect`, defaulting to `false`

2) In `channel()` method we should reconnect in two following cases:

* channel has never been opened yet. There should be some kind of internal status in the `ChromeCast` object.
* channel has been closed and `autoReconnect` is set to `true`.

3) When channel has been opened, `autoReconnect` is set to `false`, and then channel is closed, an `ChromeCastException` must be thrown about closed channel with advise on setting up `autoReconnect` property.

I hope this way it will be most flexible and we won't break backwards compatibility much. I have some doubts about `autoReconnect` default value, maybe it should be `true`.

> I had another commit making the whole thing an java.io.AutoClosable, but this really breaks the API logic because close() is not disconnect() and keeping both will confuse everyone.

I think it is fine if there is a proper documentation on the `close()` method. Please file a PR if possible.